### PR TITLE
PHOENIX-5285: Add documentation for DataSourceV2 support in the phoenix-spark connector

### PR DESCRIPTION
**Pull request by ChinmaySKulkarni**

### CODE REVIEW

**Comment by ChinmaySKulkarni** [comment]
Please review @twdsilva @dbwong

File: phoenix-spark/README.md [hunk 1]
```diff
@@ -28,27 +28,33 @@ UPSERT INTO TABLE1 (ID, COL1) VALUES (1, 'test_row_1');
 UPSERT INTO TABLE1 (ID, COL1) VALUES (2, 'test_row_2');
```
Comment (dbwong) [inline phoenix-spark/README.md]: These are not just DDL maybe reward this?

**Comment by ChinmaySKulkarni** [comment]
@dbwong addressed review comments, please take a look. The license header is there, but it won't show up when you see the readme in visual mode. You can see it exists from the [raw](https://raw.githubusercontent.com/apache/phoenix-connectors/9c163c42c9bc3074955851e77b2832b9ad3e776b/phoenix-spark/README.md) version.

**Comment by ChinmaySKulkarni** [comment]
Thanks for the review @dbwong I will squash and commit this.

**Pull request closed without merging**